### PR TITLE
LPS 153279 Style books token definition of type Boolean

### PR DESCRIPTION
**Pull request by chileces**
Hello mr Thien, this is a PR to fix the issue according to the ticket https://issues.liferay.com/browse/LPS-153279
please help to review and leave your comment.
Solution:
According to the API document, section Matching CSS Variables to Style Book Tokens:
https://learn.liferay.com/dxp/latest/en/site-building/site-appearance/style-books/developer-guide/style-book-token-definitions.html
the mapping must match with the frontend-token-definition.json structure.
also have to change the mapping from (String,String) to (String,Object)

### CODE REVIEW

File: modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java [hunk 1]
```diff
@@ -66,30 +68,15 @@ public Collection<ScopedCSSVariables> getScopedCSSVariablesCollection(
...
 			new ScopedCSSVariables() {
 
-				public Map<String, String> getCSSVariables() {
-					Map<String, String> cssVariables = new HashMap<>();
+				public Map<String, Object> getCSSVariables() {
+					Map<String, Object> cssVariables = new HashMap<>();
 
 					try {
-						JSONObject frontendTokensValuesJSONObject =
+						JSONObject jsonObject =
```
Comment (thienquach87) [inline modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java]: set variable name(jsonObject) to a meaning name

File: modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java [hunk 2]
```diff
@@ -104,6 +91,57 @@ public String getScope() {
...
+					FrontendToken.Type type = FrontendToken.Type.parse(
+						variableType);
+
+					if (type == FrontendToken.Type.BOOLEAN) {
+						defaultValue = jsonObject.getBoolean("defaultValue");
+					}
+					else if (type == FrontendToken.Type.INT) {
+						defaultValue = jsonObject.getInt("defaultValue");
+					}
+					else if (type == FrontendToken.Type.DOUBLE) {
```
Comment (thienquach87) [inline modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java]: you can shorten code by below lines
if(Arrays.asList(FrontendToken.Type.values()).contains(type)){
	defaultValue = (Object) jsonObject.get("defaultValue");
}else {
						throw new JSONException(
							"Unsupported frontend token type " +
								type.toString());
					}

File: modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java [hunk 3]
```diff
@@ -104,6 +91,57 @@ public String getScope() {
...
+					if (!JSONUtil.isEmpty(mappingsJSONArray)) {
+						for (int i = 0; i < mappingsJSONArray.length(); i++) {
+							String cssVariableName =
+								mappingsJSONArray.getJSONObject(
+									i
+								).getString(
+									"value"
+								);
+
+							cssVariables.put(cssVariableName, defaultValue);
```
Comment (thienquach87) [inline modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java:140]: cssVariables.put(cssVariableName, (Object) defaultValue);

**Comment by thienquach87** [comment]
You should update on the Test files such as: ScopedCSSVariablesTopHeadDynamicIncludeTest.java... and so on

File: modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java (hunk 2 above)
Comment (chileces) [reply modules/apps/style-book/style-book-service/src/main/java/com/liferay/style/book/internal/frontend/css/variables/StyleBookScopedCSSVariablesProvider.java]: i fixed it. thank you

**Comment by thienquach87** [comment]
LGTM

**Pull request closed without merging**